### PR TITLE
Convert packetIO buffer to an interface

### PR DESCRIPTION
**Pull request by OrlandoCo**
#### Description
PacketIO buffer interface will allow to use custom buffer for rtp/rtcp streams

### CODE REVIEW

File: packetio/buffer.go [hunk 1]
```diff
@@ -10,11 +10,34 @@ import (
...
+type Buffer interface {
+	Write(packet []byte) (int, error)
+	Read(packet []byte) (int, error)
+	Close() (err error)
+	Count() int
+	SetLimitCount(limit int)
+	Size() int
+	SetLimitSize(limit int)
+	SetReadDeadline(t time.Time) error
+	SetBufferMetadata(t BufferPacketType, ssrc uint32)
```
Comment (Sean-Der) [inline packetio/buffer.go:35]: Do we need the metadata in the base class? We may also be handling ICE and DTLS traffic.

File: packetio/buffer.go [hunk 2]
```diff
@@ -10,11 +10,34 @@ import (
...
+	RTPBufferPacket BufferPacketType = 1
+	// RTCPBufferPacket indicates the buffer that is handling RTCP packets
+	RTCPBufferPacket BufferPacketType = 2
+)
+
 var errPacketTooBig = errors.New("packet too big")
 
-// Buffer allows writing packets to an intermediate buffer, which can then be read form.
+// Buffer is an interface to implement a writing/reading packet buffer.
+type Buffer interface {
```
Comment (Sean-Der) [inline packetio/buffer.go:26]: Nice! Glad to see this was such an easy change. 

Can/should this just embed stdlib interfaces? Read/Write/Close + Deadline I think might already be something.

File: packetio/buffer.go (hunk 1 above)
Comment (OrlandoCo) [reply packetio/buffer.go:35]: I could not find anothwer way to inform the buffer which SSRC and packet type is handling

Comment (OrlandoCo) [reply packetio/buffer.go:35]: @Sean-Der  You can see what I mean in https://github.com/pion/srtp/pull/126/files

File: packetio/buffer.go (hunk 2 above)
Comment (OrlandoCo) [reply packetio/buffer.go:26]: I can set the `io.ReadWriteCloser` but not sure for the other ones

**Comment by OrlandoCo** [comment]
@Sean-Der I removed the metadata from buffer and set stdlib interfaces were possible, to understand the changes please also review https://github.com/pion/srtp/pull/126

**Pull request closed without merging**